src/firefly/java/edu/caltech/ipac/firefly/server/query/lc/IrsaLightCurveHandler.java

```diff
@@ -173,22 +175,25 @@ boolean isPost(PeriodogramAPIRequest url) {
         return url.getLcSource().indexOf("http") < 0;
     }
 
-    void insertPostParams(PeriodogramAPIRequest request, MultiPartPostBuilder posBuilder) {
-        //        rootApiUrl+= "alg=ls"+"&"; //Optional
-        // Col definition time, data
+    /**
+     * create the HttpServiceInput to call Http's post method
+     * @param request
+     * @return
+     */
+    private HttpServiceInput getHttpInput(PeriodogramAPIRequest request) {
+        HttpServiceInput httpInputs = new HttpServiceInput();
+
         String src = getSourceFile(request.getLcSource());
-        posBuilder.addFile("upload", new File(src));
+        httpInputs.setFile("upload", new File(src));
```

Comment:
Logic should be here.  This is where you take the JSON request, convert to a java Request, call the manager to save the table, then use it as the source to your post request.
```
TableServerRequest sreq = QueryUtil.convertToServerRequest(src);
sreq.setPageSize(Integer.MAX_VALUE);               // ensure you're getting the full table
sreq.removeParam(INCL_COLUMNS);                  // ensure you're getting all of the columns
File tmpOut = new File("whateveryouwanttocallit");
new SearchManager().save(new FileOutputStream(tmpOut), sreq);
httpInputs.setFile("upload", tmpOut);

Reply:
Well, I used setPageSize but did not work for me.  I still got 100 rows.  I can try one more time with sreq.removeParam that I did not use it along with setPageSize.

Reply:
@loitly 
Initially you suggested me to put the logic in LightCurveProcessor.  So I did as what you suggested.    Without looking at the implementation details, it is hard to judge the best approach.  If use one path as you both wanted, put logic here is making more sense.  I made all the related changes.  I tested  and found that overwriting file is fine.  So I removed the getFileName method as you commented.  I checked the tables for both filtered and none filtered, from Gator and from UI, all work correct.   Please review it again.  Since I removed some methods, I hope that I did not mess up things.   Thanks!